nbextensions/bulkImportCell/jobManager.js

```diff
@@ -0,0 +1,250 @@
+            ['created', 'estimating'].forEach((status) => {
+                if (statusIx[status]) {
+                    delete statusIx[status];
+                    statusIx.queued = 1;
+                }
+            });
+
+            const jobLabelString = String.arrayToEnglish(
+                Object.keys(statusIx)
+                    .sort()
+                    .map((status) => Jobs.jobLabel({ status: status }))
+            );
+            const jobString = jobList.length === 1 ? '1 job' : jobList.length + ' jobs';
+            const ucfirstAction = String.capitalize(action);
+            return {
+                title: `${ucfirstAction} ${jobLabelString} jobs`,
+                body: div([
+                    p([
+                        `${ucfirstAction}ing all ${jobLabelString} jobs will terminate the processing of ${jobString}. `,
+                        'Any output objects already created will remain in your narrative and can be removed from the Data panel.',
+                    ]),
+                    statusList.includes('error')
+                        ? p(
+                              'Please note that jobs are rerun using the same parameters. Any jobs that failed due to issues with the input, such as misconfigured parameters or corrupted input data, are likely to throw the same errors when run again.'
+                          )
+                        : '',
+                    p(`${ucfirstAction} all ${jobLabelString} jobs?`),
+                ]),
+            };
+        }
+
+        const jobCommand = {
+            cancel: 'request-job-cancellation',
+            retry: 'request-job-rerun',
+        };
+
+        /**
+         *
+         * @param {object} args with keys
+         *                 action: the action to perform -- cancel or retry
+         *                 jobId: job to perform it on
+         * @returns {boolean}
+         */
+        function executeActionOnJobId(args) {
+            const { action, jobId } = args;
+            const jobState = model.getItem(`exec.jobs.byId.${jobId}`);
+
+            if (jobState && Jobs.canDo(action, jobState)) {
+                bus.emit(jobCommand[action], {
+                    jobId: jobId,
+                });
+                return true;
+            }
+            return false;
+        }
+
+        /**
+         *
+         * @param {object} args with keys
+         *      action:        {string} action to perform (cancel or retry)
+         *      statusList:    {array}  list of statuses to perform it on
+         *      validStatuses: {array}  list of statuses that it is valid to perform the action on
+         *
+         * @returns {Promise} that resolves to false if there is some error with the input or
+         * if the user cancels the batch action. If the users confirms the action, the appropriate
+         * message will be emitted by the bus.
+         */
+        async function executeActionByJobStatus(args) {
```

Comment:
For the cancel/retry by status buttons; these now pop a modal up so the user can choose whether or not to proceed. If there are no jobs in the appropriate state, nothing happens.